File: src/main/java/com/catboyranch/roborancher/commands/CSay.java
package com.catboyranch.roborancher.commands;

import com.catboyranch.roborancher.utils.CommandArgument;
import com.catboyranch.roborancher.utils.Utils;
import com.catboyranch.roborancher.utils.RoleType;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import com.catboyranch.roborancher.RoboRancher;
import com.catboyranch.roborancher.Server;

public class CSay extends CommandBase{

    public CSay(RoboRancher rancher) {
        super(rancher);
        name = "say";
        description = "Speak catboy!";
        allowedRoles = Utils.asArray(RoleType.ADMIN);
    }

    @Override
    public void run(String cmd, Server server, MessageReceivedEvent event, CommandResult result, CommandArgument... args) {
        String message = event.getMessage().getContentRaw();
        message = message.replace(server.getConfig().getCmdPrefix() + name, "");
        event.getMessage().delete().queue();
        event.getChannel().sendMessage(message).queue();
        result.successQuiet();
    }
}

File: src/main/kotlin/com/catboyranch/roborancher/commands/CSay.kt
package com.catboyranch.roborancher.commands

import com.catboyranch.roborancher.ChannelUtils
import com.catboyranch.roborancher.RoboRancher
import com.catboyranch.roborancher.RoleUtils
import com.catboyranch.roborancher.Server
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class CSay(rancher: RoboRancher): CommandBase(rancher) {
    override val name = "say"
    override val description = "Speak catboy!"
    override val allowedRoles = arrayOf(RoleUtils.RoleType.ADMIN)

    override fun run(cmd: String, server: Server, event: MessageReceivedEvent, result: CommandResult, vararg args: CommandArgument) {
        val message = event.message.contentRaw.replace(server.config.getData().cmdPrefix + name, "")
        ChannelUtils.deleteMessage(event.message)
        event.channel.sendMessage(message).queue()
        result.successQuiet()
    }
}